File: Genral_All_Controls/Custom_OnDemand_MCCB_as_GridEditor/Custom_OnDemand_MCCB_as_GridEditor/CustomMCCBEditor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Telerik.WinControls.UI;

namespace _1408634
{
    public partial class CustomMCCBEditor : UserControl
    {
        public CustomMCCBEditor()
        {
            InitializeComponent();
            this.radGridView1.AllowAddNewRow = false;
            this.radGridView1.AutoSizeColumnsMode = GridViewAutoSizeColumnsMode.Fill;
            this.radPopupEditor1.DropDownStyle = Telerik.WinControls.RadDropDownStyle.DropDown;
            this.radGridView1.ReadOnly = true;
            this.radGridView1.CurrentRowChanged += RadGridView1_CurrentRowChanged;
            this.TextBox.KeyDown += TextBox_KeyDown;
            this.radGridView1.CellDoubleClick += RadGridView1_CellDoubleClick;
        }

        private void RadGridView1_CellDoubleClick(object sender, GridViewCellEventArgs e)
        {
            this.PopupEditor.PopupEditorElement.ClosePopup();
        }

        private void TextBox_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Enter)
            {
                this.PopupEditor.PopupEditorElement.ClosePopup();
            }
        }

        private void RadGridView1_CurrentRowChanged(object sender, CurrentRowChangedEventArgs e)
        {
         

        }

        public object Value
        {
            get
            {
                if (this.radGridView1.CurrentRow != null && this.radGridView1.CurrentRow is GridViewDataRowInfo)
                {
                    return this.radGridView1.CurrentRow.Cells[0].Value;
                }
                return null;
            }
        }
        public RadTextBoxElement TextBox
        {
            get
            {
                return this.radPopupEditor1.TextBoxElement;
            }
        }
        public RadGridView GridControl
        {
            get
            {
                return this.radGridView1;
            }
        }
        public RadPopupEditor PopupEditor
        {
            get
            {
                return this.radPopupEditor1;
            }
        }

    }
}

File: Genral_All_Controls/Custom_OnDemand_MCCB_as_GridEditor/Custom_OnDemand_MCCB_as_GridEditor/CustomMCCBEditor.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports Telerik.WinControls.UI

Namespace _1408634
	Partial Public Class CustomMCCBEditor
		Inherits UserControl

		Public Sub New()
			InitializeComponent()
			Me.radGridView1.AllowAddNewRow = False
			Me.radGridView1.AutoSizeColumnsMode = GridViewAutoSizeColumnsMode.Fill
			Me.radPopupEditor1.DropDownStyle = Telerik.WinControls.RadDropDownStyle.DropDown
			Me.radGridView1.ReadOnly = True
			AddHandler Me.radGridView1.CurrentRowChanged, AddressOf RadGridView1_CurrentRowChanged
			AddHandler Me.TextBox.KeyDown, AddressOf TextBox_KeyDown
			AddHandler Me.radGridView1.CellDoubleClick, AddressOf RadGridView1_CellDoubleClick
		End Sub

		Private Sub RadGridView1_CellDoubleClick(ByVal sender As Object, ByVal e As GridViewCellEventArgs)
			Me.PopupEditor.PopupEditorElement.ClosePopup()
		End Sub

		Private Sub TextBox_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs)
			If e.KeyCode = Keys.Enter Then
				Me.PopupEditor.PopupEditorElement.ClosePopup()
			End If
		End Sub

		Private Sub RadGridView1_CurrentRowChanged(ByVal sender As Object, ByVal e As CurrentRowChangedEventArgs)


		End Sub

		Public ReadOnly Property Value() As Object
			Get
				If Me.radGridView1.CurrentRow IsNot Nothing AndAlso TypeOf Me.radGridView1.CurrentRow Is GridViewDataRowInfo Then
					Return Me.radGridView1.CurrentRow.Cells(0).Value
				End If
				Return Nothing
			End Get
		End Property
		Public ReadOnly Property TextBox() As RadTextBoxElement
			Get
				Return Me.radPopupEditor1.TextBoxElement
			End Get
		End Property
		Public ReadOnly Property GridControl() As RadGridView
			Get
				Return Me.radGridView1
			End Get
		End Property
		Public ReadOnly Property PopupEditor() As RadPopupEditor
			Get
				Return Me.radPopupEditor1
			End Get
		End Property

	End Class
End Namespace
